metaflow/plugins/cards/card_datastore.py

```diff
@@ -89,10 +108,21 @@ def _make_path(self, base_pth, pathspec=None):
         sysroot = base_pth
 
         if pathspec is not None:
+            # since most cards are at a task level there will always be 4 non-none values returned
             flow_name, run_id, step_name, task_id = path_spec_resolver(pathspec)
-
-        # For task level cards the flow_name and run_id and task_id are required
-        if flow_name is not None and run_id is not None and task_id is not None:
+        if abs_version(self._metaflow_version) >= 2780:
```

Comment:
Rather than depending explicitly on metaflow-version, we can also first check for the card's existence in the new location and fall back on the old location if the card doesn't exist. There is no guarantee that this PR will ship with Metaflow 2.7.8.